src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/NotificationsService.cs

```diff
@@ -26,27 +26,23 @@ public class NotificationsService : INotificationsService
 
         public async Task NotifySubscribersIfApplicable(params Guid[] releaseIds)
         {
-            var releasesToNotify = _context.Releases
-                .Include(r => r.ReleaseStatuses)
-                .Where(r => releaseIds.Contains(r.Id))
-                .ToList()
-                .Where(r => r.NotifySubscribers)
-                .ToList();
+            var releasesToNotify = await _context.Releases
+                .Where(r => releaseIds.Contains(r.Id) && r.NotifySubscribers && r.NotifiedOn == null)
```

Comment:
I can't see any harm in testing the NotifiedOn date being null here, but in reality should we ever be able to reach this bit of code when NotifiedOn is *not* null?

Either way, happy for it to stay this way, just curious.

Reply:
That's correct, I don't think we should ever be able to reach here with `NotifiedOn` having a not null value.

I was copying the previous behaviour of  the previous get accessor `Release.NotifySubscribers` when it depended on `ReleaseStatsu`:

```
        public bool NotifySubscribers
        {
            get
            {
                var status = ReleaseStatuses
                           .OrderBy(rs => rs.Created)
                           .LastOrDefault();
                return status != null
                       && status.NotifySubscribers
                       && status.NotifiedOn == null;
            }
        }
```

I think we can get rid of this check.

I suppose it's useful to stop subscribers being emailed multiple times if we were to ever force a release through the publishing process more than once rather than amending, but it seems unlikely we'd need to do that and it can still be prevented by altering `NotifySubscribers` to false.

Reply:
> That's correct, I don't think we should ever be able to reach here with `NotifiedOn` having a not null value.
> 
> I was copying the previous behaviour of the previous get accessor `Release.NotifySubscribers` when it depended on `ReleaseStatus`:
> 
> ```
>         public bool NotifySubscribers
>         {
>             get
>             {
>                 var status = ReleaseStatuses
>                            .OrderBy(rs => rs.Created)
>                            .LastOrDefault();
>                 return status != null
>                        && status.NotifySubscribers
>                        && status.NotifiedOn == null;
>             }
>         }
> ```
> 
> I think we can get rid of the `NotifiedOn == null` check.
> 
> I suppose it's useful to stop subscribers being emailed multiple times if we were to ever force a release through the publishing process more than once rather than amending, but it seems unlikely we'd need to do that and it can still be prevented by altering `NotifySubscribers` to false.

Reply:
Sure, happy either way thanks!